### PR TITLE
Update scalajs-react / react / Scala.JS

### PR DESCRIPTION
**Pull request by alexarchambault**

### CODE REVIEW

File: modules/core/js/src/main/scala/coursier/core/compatibility/package.scala [hunk 1]
```diff
@@ -34,10 +44,10 @@ package object compatibility {
   lazy val DOMParser = newFromXmlDomOrGlobal("DOMParser")
   lazy val XMLSerializer = newFromXmlDomOrGlobal("XMLSerializer")
   lazy val sax =
-    if (js.isUndefined(g.sax))
-      g.require("sax")
+    if (js.isUndefined(globalObject.sax))
+      globalObject.require("sax")
     else
-      g.sax
+      globalObject.sax
```
Comment (sjrd) [inline modules/core/js/src/main/scala/coursier/core/compatibility/package.scala:50]: If you do things this way, you'll *still* have the problem that "`require` is not defined". You're going out of your way to reproduce the behavior of Scala.js 0.6.x, which is the cause of #1618.

You should take full advantage of the behavior of Scala.js 1.x instead, as follows:
```scala
lazy val sax =
  if (js.typeOf(g.sax) == "undefined")
    g.require("sax")
  else
    g.sax
```
Likewise, you should inline `newFromXmlDomOrGlobal` at the two places where it's used to be able to use the same pattern:
```scala
lazy val DOMParser = {
  val defn =
    if (js.typeOf(g.DOMParser) == "undefined") g.require("xmldom").DOMParser
    else g.DOMParser
  js.Dynamic.newInstance(defn)()
}
```
and copy-paste for `XMLSerializer`.

If you can't stand the duplication, you can abstract away *some* of it as follows:
```scala
def newFromXmlDomOrGlobal(typeOf: String, globalDefn: => js.Dynamic, xmldomName: String) = {
  val defn =
    if (typeOf == "undefined") g.require("xmldom").selectDynamic(xmldomName)
    else globalDefn
  js.Dynamic.newInstance(defn)()
}
```
which you can use as
```scala
lazy val DOMParser = newFromXmlDomOrGlobal(js.typeOf(g.DOMParser), g.DOMParser, "DOMParser")
lazy val XMLSerializer = newFromXmlDomOrGlobal(js.typeOf(g.XMLSerializer), g.XMLSerializer, "XMLSerializer")
```

File: modules/tests/js/src/test/scala/coursier/Platform.scala [hunk 2]
```diff
@@ -15,12 +15,21 @@ object Platform {
...
+
   /** Available if we're running on node, and package xhr2 is installed */
   lazy val xhr =
-    if (js.isUndefined(g.XMLHttpRequest))
-      g.require("xhr2")
+    if (js.isUndefined(globalObject.XMLHttpRequest))
+      globalObject.require("xhr2")
     else
-      g.XMLHttpRequest
+      globalObject.XMLHttpRequest
```
Comment (sjrd) [inline modules/tests/js/src/test/scala/coursier/Platform.scala:32]: Of course, the same story applies here, and everywhere else where this pattern comes up.

File: modules/util/js/src/main/scala/coursier/util/WebPageCompatibility.scala [hunk 3]
```diff
@@ -6,10 +6,19 @@ import scala.scalajs.js.Dynamic.{global => g}
...
+      g.global
+    else
+      // In all other well-known environment, we can use the global `this`
+      js.special.fileLevelThis.asInstanceOf[js.Dynamic]
+
   // FIXME Won't work in the browser
   lazy val cheerio = g.require("cheerio")
 
-  lazy val jqueryAvailable = !js.isUndefined(g.$)
+  lazy val jqueryAvailable = !js.isUndefined(globalObject.$)
```
Comment (sjrd) [inline modules/util/js/src/main/scala/coursier/util/WebPageCompatibility.scala:21]:
```suggestion
  lazy val jqueryAvailable = js.typeOf(g.$) != "undefined"
```

File: modules/util/js/src/main/scala/coursier/util/WebPageCompatibility.scala [hunk 4]
```diff
@@ -18,7 +27,7 @@ private[coursier] abstract class WebPageCompatibility {
     // getting weird "maybe a wrong Dynamic method signature" errors when trying to factor that more
 
     if (jqueryAvailable)
-        g.$("<div></div>").html(page).find("a").each({ self: js.Dynamic =>
+        globalObject.$("<div></div>").html(page).find("a").each({ self: js.Dynamic =>
```
Comment (sjrd) [inline modules/util/js/src/main/scala/coursier/util/WebPageCompatibility.scala:30]:
```suggestion
        g.$("<div></div>").html(page).find("a").each({ self: js.Dynamic =>
```

**Comment by alexarchambault** [comment]
Thanks for the review, @sjrd!

**Pull request merged**